rules/CodeQuality/Rector/Class_/StaticToSelfStaticMethodCallOnFinalClassRector.php

```diff
@@ -73,54 +70,6 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Class_
     {
-        if (! $node->isFinal() && FeatureFlags::treatClassesAsFinal($node) === false) {
-            return null;
-        }
-
-        $hasChanged = false;
-        $scope = ScopeFetcher::fetch($node);
-        $classReflection = $scope->getClassReflection();
-
-        if (! $classReflection instanceof ClassReflection) {
-            return null;
-        }
-
-        $this->traverseNodesWithCallable($node->stmts, function (Node $subNode) use (&$hasChanged, $classReflection): ?StaticCall {
-            if (! $subNode instanceof StaticCall) {
-                return null;
-            }
-
-            if (! $this->isName($subNode->class, ObjectReference::STATIC)) {
-                return null;
-            }
-
-            // skip dynamic method
-            if (! $subNode->name instanceof Identifier) {
-                return null;
-            }
-
-            $methodName = (string) $this->getName($subNode->name);
-
-            if (! $classReflection->hasNativeMethod($methodName)) {
-                return null;
-            }
-
-            $extendedMethodReflection = $classReflection->getNativeMethod($methodName);
-
-            // avoid overlapped change
-            if (! $extendedMethodReflection->isStatic()) {
-                return null;
-            }
-
-            $hasChanged = true;
-            $subNode->class = new Name('self');
-            return $subNode;
-        });
-
-        if ($hasChanged) {
-            return $node;
-        }
-
-        return null;
+        return $this->convertStaticToSelfRector->refactor($node);
```

Comment:
Awesome 😎

Reply:
The new rule already registered at set

https://github.com/rectorphp/rector-src/pull/7161/files#diff-13717e603efd6359b875e4e07c168ce004e8e9434fef929887c734014c018e01

just ensuring user that manually register old rules not got invalid apply rule duplication check because call rector rule as a service.

Reply:
sure, here the PR:

- https://github.com/rectorphp/rector-src/pull/7165